src/components/Property/Types/SelectionProperty.tsx

```diff
@@ -1,30 +1,38 @@
 import { MultiSelect } from '@mantine/core';
 
-import { ConcretePropertyBaseProps } from '../types';
+import {
+  useGetPropertyDescription,
+  useGetSelectionPropertyValue,
+  usePropListeningState
+} from '@/api/hooks';
+import { PropertyProps } from '@/components/Property/types';
 
-interface Props extends ConcretePropertyBaseProps {
-  setPropertyValue: (newValue: string[]) => void;
-  additionalData: {
-    Options: string[];
-  };
-  value: string[];
-}
+export function SelectionProperty({ uri }: PropertyProps) {
+  const [value, setValue] = useGetSelectionPropertyValue(uri);
+  const { value: currentValue, setValue: setCurrentValue } = usePropListeningState<
+    string[] | undefined
+  >(value);
+
+  const description = useGetPropertyDescription(uri);
+
+  if (!value || !description || currentValue === undefined) {
+    return <></>;
+  }
+
+  const options = description.additionalData.Options as string[];
+
+  function handleChange(newValue: string[]) {
+    setValue(newValue);
+    setCurrentValue(newValue);
+  }
 
-export function SelectionProperty({
-  name,
-  disabled,
-  setPropertyValue,
-  value,
-  additionalData
-}: Props) {
-  const options = additionalData.Options;
   return (
     <MultiSelect
       aria-label={`${name} multi-select`}
-      disabled={disabled}
+      disabled={description.additionalData.isReadOnly}
```

Comment:
Same here!  It's really an issue that this does not lead to an error. We should look at the type handling for `additionalData`

```suggestion
      disabled={description.metaData.isReadOnly}
```